platform/entanglement/test/services/CopyServiceSpec.js

```diff
@@ -266,6 +285,20 @@ define(
                     });
                 });
 
+                it("throws an expection when performed on invalid inputs", function () {
+                    var copyService =
+                        new CopyService(mockQ, creationService, policyService);
+
+                    function perform() {
+                        copyService.perform(object, newParent);
+                    }
+
+                    policyService.allow.andReturn(true);
```

Comment:
A better perform unit test would be to `spyOn(copyService, 'validate')` and ensure that it is called with the correct arguments and that the return value of `validate` is actually what is being tested, and not the return value of policy service.